plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/models/ResourceCache.java

```diff
@@ -0,0 +1,177 @@
+/*******************************************************************************
+ * Copyright (c) 2016 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v1.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package org.jboss.tools.openshift.internal.ui.models;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.eclipse.osgi.util.NLS;
+import org.jboss.tools.openshift.internal.core.Trace;
+
+import com.openshift.restclient.model.IProject;
+import com.openshift.restclient.model.IResource;
+
+public class ResourceCache {
+	private Map<String, Map<String, IResource>> cache = new HashMap<String, Map<String, IResource>>();
+
+	public void dispose() {
+		flush();
+	}
+
+	public void flush() {
+		synchronized (cache) {
+			cache.clear();
+		}
+	}
+
+	public void flush(String namespace) {
+		synchronized (cache) {
+			cache.remove(namespace);
+		}
+	}
+
+	/**
+	 * Key for caching an object
+	 * 
+	 * @param resource
+	 * @return
+	 */
+	private String getCacheKey(IResource resource) {
+		return getCacheKey(resource.getKind(), resource.getName());
+	}
+
+	private String getCacheKey(String kind, String name) {
+		return NLS.bind("{0}/{1}", new Object[] { kind, name });
```

Comment:
It don't think it is heavily used. Perhaps hundreds of times each time a resource. What makes you think otherwise? 
In other news, I agree it's overkill.